window_manager/src/XWWindow.cpp

```diff
@@ -0,0 +1,53 @@
+#include <future>
+
+#include <core/XWWindow.hpp>
+#ifdef __linux__
+#include "XWindow.hpp"
+#elif __WIN32
+#include "WinWindow.hpp"
+#endif
+
+namespace rcbe::core {
+XWWindow::XWWindow(WindowConfig &&config_, Display* root_display, int screen_number, const Window& root_window, const Atom& delete_msg)
+:
+impl_ { std::make_shared<XWindow>(std::move(config_), root_display, screen_number, root_window, delete_msg) }
+{
+
+}
+
+XWWindow::~XWWindow() {
+
```

Comment:
Check if running and kill

Reply:
Underlying class already has this functionality in it's destructor